src/thermo/CoverageDependentSurfPhase.cpp

```diff
@@ -0,0 +1,772 @@
+    scale(m_entropy.begin(), m_entropy.end(), sr, 1.0/GasConstant);
+    if (m_has_ref_coverage) {
+        double tmp = -log(m_theta_ref);
+        for (size_t k = 0; k < m_kk; k++) {
+            sr[k] -= tmp;
+        }
+    }
+}
+
+void CoverageDependentSurfPhase::getCp_R(doublereal* cpr) const
+{
+    _updateThermo();
+    scale(m_heatcapacity.begin(), m_heatcapacity.end(), cpr, 1.0/GasConstant);
+}
+
+void CoverageDependentSurfPhase::getGibbs_RT(doublereal* grt) const
+{
+    _updateThermo();
+    scale(m_chempot.begin(), m_chempot.end(), grt, 1.0/RT());
+    if (m_has_ref_coverage) {
+        double tmp = -log(m_theta_ref);
+        for (size_t k = 0; k < m_kk; k++) {
+            grt[k] += tmp;
+        }
+    }
+}
+
+void CoverageDependentSurfPhase::getPureGibbs(doublereal* g) const
+{
+    getGibbs_RT(g);
+    for (size_t k = 0; k < m_kk; k++) {
+        g[k] *= RT();
+    }
+}
+
+void CoverageDependentSurfPhase::getStandardChemPotentials(doublereal* mu0) const
+{
+    _updateThermo();
+    copy(m_chempot.begin(), m_chempot.end(), mu0);
+    if (m_has_ref_coverage) {
+        double tmp = RT() * -log(m_theta_ref);
+        for (size_t k = 0; k < m_kk; k++) {
+            mu0[k] += tmp;
+        }
+    }
+}
+
+// Functions calling partial molar thermodyanmic properties----------------
+
+void CoverageDependentSurfPhase::getPartialMolarEnthalpies(doublereal* hbar) const
+{
+    _updateThermo();
+    copy(m_enthalpy.begin(), m_enthalpy.end(), hbar);
+}
+
+void CoverageDependentSurfPhase::getPartialMolarEntropies(doublereal* sbar) const
+{
+    _updateThermo();
+    copy(m_entropy.begin(), m_entropy.end(), sbar);
+    if (m_has_ref_coverage) {
+        for (size_t k = 0; k < m_kk; k++) {
+            sbar[k] -= GasConstant * log(std::max(concentration(k) * size(k)/m_n0, 
+                SmallNumber) / m_theta_ref);
+        }
+    } else {
+        for (size_t k = 0; k < m_kk; k++) {
+            sbar[k] -= GasConstant * log(std::max(concentration(k) * size(k)/m_n0, 
+                SmallNumber));
+        }
+    }
+}
+
+void CoverageDependentSurfPhase::getPartialMolarCp(doublereal* cpbar) const
+{
+    _updateThermo();
+    copy(m_heatcapacity.begin(), m_heatcapacity.end(), cpbar);
+}
+
+
+void CoverageDependentSurfPhase::getChemPotentials(doublereal* mu) const
+{
+    _updateThermo();
+    copy(m_chempot.begin(), m_chempot.end(), mu);
+    if (m_has_ref_coverage) {
+        for (size_t k = 0; k < m_kk; k++) {
+            mu[k] += RT() * log(std::max(concentration(k) * size(k)/m_n0, 
+                SmallNumber) / m_theta_ref);
+        }
+    } else {
+        for (size_t k = 0; k < m_kk; k++) {
+            mu[k] += RT() * log(std::max(concentration(k) * size(k)/m_n0, 
+                SmallNumber));
+        }
+    }
+}
+
+// Functions calculating mixture thermodyanmic properties--------------------------
+
+doublereal CoverageDependentSurfPhase::enthalpy_mole() const
+{
+    if (m_n0 <= 0.0) {
+        return 0.0;
+    }
```

Comment:
This is kind of a strange case to check -- I don't see it being done anywhere else.